doc/source/index.rst

```diff
@@ -68,6 +68,7 @@ Documentation
     composites
     resample
     enhancements
+    readers_formats
```

Comment:
I think the location of the document in the TOC is the only thing that I disagree with in this PR. It feels like this should either go right after "reading" or go after the dev_guide/index in the sense that it is an "extra" document. I think I prefer the latter.